grid/chat/channels/maps/slack.suma

```diff
@@ -0,0 +1,212 @@
+profile = "chat/channels@1.0"
+provider = "slack"
+
+"""
+GetChannels map
+"""
+map GetChannels {
+  types = call typesToQuery(types = input.types)
+
+  http GET "/conversations.list" {
+    request "application/x-www-form-urlencoded" {
+      query {
+        cursor = input.page
+        exclude_archived = true
+        limit = input.limit
+        types = types
+      }
+
+      headers {
+        "Authorization" = `Bearer ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {   
+      error = call MapSlackError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      ) if (!body.ok)
+
+      return map error if (!body.ok) error
+
+      channels = call MapChannels(channels = body.channels)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+      result.channels = channels
+
+      set if (body.has_more) {
+        result.nextPage = body.response_metadata.next_cursor
+      }
+
+      set if (rateLimit) {
+        result.rateLimit = rateLimit
+      }
+
+      map result result
+    }
+
+    response {
+      error = call MapSlackError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+operation MapChannels {
+  channels = []
+
+  channels = args.channels.map(channel => {
+    const resultChannel = {
+      id: channel.id,
+      createdAt: channel.created,
+    };
+
+    if (channel.name) {
+      resultChannel.name = channel.name;
+    }
+
+    if (channel.topic.value && channel.topic.value !== '') {
+      resultChannel.description = channel.topic.value;
+    }
+
+    if (channel.purpose.value && channel.purpose.value !== '') {
+      resultChannel.description = 
+        resultChannel.description === undefined 
+        ? channel.purpose.value
+        : `${resultChannel.description} - ${channel.purpose.value}`;
+    }
```

Comment:
I am not 100% sure about this creative concatenation of description and purpose, personally I'd use purpose only as a fallback when description is not present.

Reply:
Check out snapshots, I think it's useful, but I don't have strong opinion on this as well.